hal/targets/hal/TARGET_ONSEMI/TARGET_NCS36510/sleep_api.c

```diff
@@ -0,0 +1,92 @@
+/**
+ *******************************************************************************
+ * @file    sleep_api.c
+ * @brief Implementation of a sleep functionality
+ * @internal
+ * @author  ON Semiconductor
+ * $Rev:  $
+ * $Date: $
+ ******************************************************************************
+ * @copyright (c) 2015 ON Semiconductor. All rights reserved.
+ * ON Semiconductor is supplying this software for use with ON Semiconductor
+ * processor based microcontrollers only.
+ *
+ * THIS SOFTWARE IS PROVIDED "AS IS".  NO WARRANTIES, WHETHER EXPRESS, IMPLIED
+ * OR STATUTORY, INCLUDING, BUT NOT LIMITED TO, IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE APPLY TO THIS SOFTWARE.
+ * ON SEMICONDUCTOR SHALL NOT, IN ANY CIRCUMSTANCES, BE LIABLE FOR SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, FOR ANY REASON WHATSOEVER.
+ * @endinternal
+ *
+ * @ingroup sleep
+ *
+ * @details
+ * Sleep implementation TBD - Dummy function is fine for first release
+ *
+ */
+
+#if DEVICE_SLEEP
+#include "sleep.h"
+#include "sleep_api.h"
+#include "cmsis_nvic.h"
+
+void mbed_enter_sleep(sleep_t *obj)
+{
+#if 0
```

Comment:
I removed the dead code in this file. The sleep API is yet to be updated from our side. We will need to do it.